onboarding/src/main/kotlin/io/homeassistant/companion/android/onboarding/sethomenetwork/SetHomeNetworkViewModel.kt

```diff
@@ -0,0 +1,111 @@
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.navigation.toRoute
+import dagger.hilt.android.lifecycle.HiltViewModel
+import io.homeassistant.companion.android.common.data.network.WifiHelper
+import io.homeassistant.companion.android.common.data.servers.ServerManager
+import io.homeassistant.companion.android.onboarding.sethomenetwork.navigation.SetHomeNetworkRoute
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import timber.log.Timber
+
+@HiltViewModel
+class SetHomeNetworkViewModel @VisibleForTesting constructor(
+    private val serverId: Int,
+    private val serverManager: ServerManager,
+    connectivityManager: ConnectivityManager,
+    wifiHelper: WifiHelper,
+) : ViewModel() {
+
+    @Inject
+    constructor(
+        connectivityManager: ConnectivityManager,
+        savedStateHandle: SavedStateHandle,
+        serverManager: ServerManager,
+        wifiHelper: WifiHelper,
+    ) : this(
+        serverId = savedStateHandle.toRoute<SetHomeNetworkRoute>().serverId,
+        serverManager,
+        connectivityManager,
+        wifiHelper,
+    )
+
+    /**
+     * We only focus on the SSID here to make things simpler for most of our users. Anyone that wants
+     * something more advance like using BSSID should do it later from the settings.
+     */
+    private val _currentWifiNetwork =
+        MutableStateFlow(wifiHelper.getWifiSsid()?.removeSurrounding("\"") ?: "")
+    val currentWifiNetwork = _currentWifiNetwork.asStateFlow()
+
+    val hasEthernetConnection: Boolean = connectivityManager.hasEthernetConnection()
+    val hasVPNConnection: Boolean = connectivityManager.hasVPNConnection()
+
+    private val _isUsingEthernet = MutableStateFlow(hasEthernetConnection)
+    val isUsingEthernet = _isUsingEthernet.asStateFlow()
+
+    private val _isUsingVpn = MutableStateFlow(hasVPNConnection)
+    val isUsingVpn = _isUsingVpn.asStateFlow()
+
+    fun onCurrentWifiNetworkChange(value: String) {
+        _currentWifiNetwork.value = value
+    }
+
+    fun onUsingEthernetChange(value: Boolean) {
+        _isUsingEthernet.value = value
+    }
+
+    fun onUsingVpnChange(value: Boolean) {
+        _isUsingVpn.value = value
+    }
+
+    fun onNextClick() {
+        viewModelScope.launch {
+            val currentWifiNetwork = currentWifiNetwork.value
+            serverManager.getServer(serverId)?.let { server ->
+                serverManager.updateServer(
+                    server.copy(
+                        connection = server.connection.copy(
+                            internalVpn = isUsingVpn.value,
+                            internalEthernet = isUsingEthernet.value,
+                            // We don't want to add an empty network in the list
+                            internalSsids = currentWifiNetwork.takeIf { it.isNotBlank() }
```

Comment:
This may be too niche for onboarding to support, but: #3354

Reply:
Since it is already supported we should too in this screen.